ddtrace/contrib/pytest/plugin.py

```diff
@@ -306,111 +310,132 @@ def _get_test_class_hierarchy(item):
+        if "reason" in marker.kwargs and marker.kwargs["reason"] == SKIPPED_BY_ITR
+    ]
 
-        # Parameterized test cases will have a `callspec` attribute attached to the pytest Item object.
-        # Pytest docs: https://docs.pytest.org/en/6.2.x/reference.html#pytest.Function
-        if getattr(item, "callspec", None):
-            parameters = {"arguments": {}, "metadata": {}}  # type: Dict[str, Dict[str, str]]
-            for param_name, param_val in item.callspec.params.items():
-                try:
-                    parameters["arguments"][param_name] = encode_test_parameter(param_val)
-                except Exception:
-                    parameters["arguments"][param_name] = "Could not encode"
-                    log.warning("Failed to encode %r", param_name, exc_info=True)
-            span.set_tag_str(test.PARAMETERS, json.dumps(parameters))
-
-        markers = [marker.kwargs for marker in item.iter_markers(name="dd_tags")]
-        for tags in markers:
-            span.set_tags(tags)
-        _store_span(item, span)
-
-        if coverage_enabled():
-            from ddtrace.internal.ci_visibility.coverage import cover
-
-            with cover(span, root=str(item.config.rootdir)):
-                yield
-        else:
+    if is_skipped_by_itr:
+        yield
+    else:
+        test_session_span = _extract_span(item.session)
+
+        pytest_module_item = _find_pytest_item(item, pytest.Module)
+        pytest_package_item = _find_pytest_item(pytest_module_item, pytest.Package)
+
+        test_module_span = _extract_span(pytest_package_item)
+        if pytest_package_item is not None and test_module_span is None:
+            if test_module_span is None:
+                test_module_span = _start_test_module_span(pytest_package_item)
+
+        test_suite_span = _extract_span(pytest_module_item)
+        if pytest_module_item is not None and test_suite_span is None:
+            test_suite_span = _start_test_suite_span(pytest_module_item)
+            # Start coverage for the test suite if coverage is enabled
+            if coverage_enabled():
+                _initialize(str(item.config.rootdir))
+                _coverage_start()
+
+        with _CIVisibility._instance.tracer._start_span(
+            ddtrace.config.pytest.operation_name,
+            service=_CIVisibility._instance._service,
+            resource=item.nodeid,
+            span_type=SpanTypes.TEST,
+            activate=True,
+        ) as span:
+            span.set_tag_str(COMPONENT, "pytest")
+            span.set_tag_str(SPAN_KIND, KIND)
+            span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+            span.set_tag_str(_EVENT_TYPE, SpanTypes.TEST)
+            span.set_tag_str(test.NAME, item.name)
+            span.set_tag_str(test.COMMAND, _get_pytest_command(item.config))
+            span.set_tag_str(_SESSION_ID, str(test_session_span.span_id))
+
+            if test_module_span is not None:
```

Comment:
I'm tempted to say we've already checked for `test_module_span == None` above. Maybe we should let this be some kind of exception (that we potentially catch and log)?

Reply:
The last we did above is check if is None and then start a test module span:
```
            if test_module_span is None:
                test_module_span = _start_test_module_span(pytest_package_item)
```
I wouldn't consider this a wrong state if it doesn't exist

Reply:
We looked at this offline and it turns out `test_module_span` is in fact a valid scenario when `pytest` is run in a context that doesn't have a package.